pkg/driver/docker/docker.go

```diff
@@ -363,6 +366,188 @@ func (d *dockerDriver) RunDockerDevContainer(
 		return err
 	}
 
+	if runtime.GOOS != "windows" && ((parsedConfig.ContainerUser != "" || parsedConfig.RemoteUser != "") &&
+		(parsedConfig.UpdateRemoteUserUID == nil || *parsedConfig.UpdateRemoteUserUID)) {
+		// Retrieve local user UID and GID
+		localUser, err := user.Current()
+		if err != nil {
+			return err
+		}
+		localUid := localUser.Uid
+		localGid := localUser.Gid
+
+		// Retrieve user to update
+		containerUser := parsedConfig.RemoteUser
+		if containerUser == "" {
+			containerUser = parsedConfig.ContainerUser
```

Comment:
do we need to check if `parsedConfig.ContainerUser` is ""? If so should we default to root?

Reply:
I don't think so because we would change the UID of root. I don't think it is a good thing.
According to spec it is only applicable when remoteUser or containerUser is set. So maybe I shall return in this case directly.

Reply:
Yes for sure let's not do that :) if the spec states it is only for remote or container user then I think we should return. There are definitely use cases where neither are set